debian/marquee/nginx/conf/whatwg.conf

```diff
@@ -38,3 +38,10 @@ types {
     text/plain md sh;
     video/ogg ogv; # https://www.iana.org/assignments/media-types/video/ogg
 }
+
+location ~* \.js$ {
+  types { }
+  default_type "text/javascript; charset=utf-8";
+}
+
+charset utf-8;
```

Comment:
This would not be applicable to text/javascript so you need to repeat it there?

Reply:
+location ~* \.js$ {
    +  types { }
    +  default_type "text/javascript; charset=utf-8";
    +}
    +
    +charset utf-8;

> This would not be applicable to text/javascript so you need to repeat it there?

The `charset utf-8` sets the encoding globally for all media types, so the repetition is really instead in `default_type "text/javascript; charset=utf-8"` directive — because (due to the somewhat unintuitive way that nginx config works) the global `charset` setting is ignored there, and so if the `charset=utf-8` isn’t explicitly specified there, the `.js` files will be served as just `text/javascript` without any `charset` param.

Reply:
(I read this through and think it makes sense. I'll add a comment though.)